nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -84,12 +84,20 @@ LifecycleManager::LifecycleManager()
       if (autostart_) {
         startup();
       }
-    });
+    },
+    callback_group_);
+
+  callback_group_executor_thread_ = std::thread([this]() {
```

Comment:
https://github.com/ros-planning/navigation2/blob/main/nav2_util/src/node_thread.cpp you can use the node thread here instead to reuse a bit of code. You'd have to change the constructor to take in an executor though. Might be worth your time since I think this will come up again on your project.

Reply:
do you means that i should add a constructor with arg executor in `NodeThread`  ? or create a new calss `ExecutorThread` ?

Reply:
You got it -- just a new constructor